sys/include/clk.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_clk System core clock
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       System core clock utility functions
+ */
+
+#ifndef CLK_H
+#define CLK_H
+
+#include <stdint.h>
+#include "periph_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Get the current system core clock frequency in Hz.
+ *
+ * @returns current system core clock frequency in Hz
+ */
+#if defined(CLOCK_CORECLOCK)
+static inline uint32_t sysclk(void) {
+    return CLOCK_CORECLOCK;
+}
+#else
+uint32_t sysclk(void);
```

Comment:
can we require this to be a static inline?

Reply:
On fe310, this function is not static inline. it could be but that would need an indirection to another function.

Reply:
But that wouldn't be an issue, would it? It would be IMO easier to read to have

```C
extern uint32_t sysclk_implementation(void);

static uint32_t sysclk(void)
{
    return sysclk_implementation ();
}
```

well encapsulated in the implementation rather than preprocessor conditionals in tge public header. But I guess it is a matter of personal taste :)

Reply:
>  encapsulated in the implementation rather than preprocessor conditionals in tge public header

That would move the preprocessor conditionals in _all_ cpu implementations. But I agree that when `CLOCK_CORECLOCK` is not defined, we can use an `extern` to a concrete cpu level implementation.